patches/chrome-browser-ui-toolbar-recent_tabs_sub_menu_model_unittest.cc.patch

```diff
@@ -1,12 +1,103 @@
+   constexpr ModelData kDataBeforeLoad[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+       {ui::MenuModel::TYPE_COMMAND, false},   // Recently closed
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+@@ -349,6 +351,7 @@ TEST_F(RecentTabsSubMenuModelTest,
+   // Expected menu items after tabs/windows from last session are loaded:
+   constexpr ModelData kDataAfterLoad[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+       {ui::MenuModel::TYPE_TITLE, false},     // Recently closed
+       {ui::MenuModel::TYPE_SUBMENU, true},    // <window>
+@@ -410,6 +413,7 @@ TEST_F(RecentTabsSubMenuModelTest, OtherDevices) {
+   // Expected menu items:
+   constexpr ModelData kData[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+       {ui::MenuModel::TYPE_COMMAND, false},   // Recently closed
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+@@ -446,6 +450,7 @@ TEST_F(RecentTabsSubMenuModelTest, OtherDevicesDynamicUpdate) {
+   // Expected menu items with sync disabled:
+   constexpr ModelData kDataSyncDisabled[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+       {ui::MenuModel::TYPE_COMMAND, false},   // Recently closed
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+@@ -458,6 +463,7 @@ TEST_F(RecentTabsSubMenuModelTest, OtherDevicesDynamicUpdate) {
+   // Expected menu items with sync enabled:
+   constexpr ModelData kDataSyncEnabled[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+       {ui::MenuModel::TYPE_COMMAND, false},   // Recently closed
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+@@ -477,6 +483,7 @@ TEST_F(RecentTabsSubMenuModelTest, OtherDevicesDynamicUpdate) {
+   // Expected menu items after update:
+   constexpr ModelData kDataAfterUpdate[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+       {ui::MenuModel::TYPE_COMMAND, false},   // Recently closed
+       {ui::MenuModel::TYPE_SEPARATOR, true},  // <separator>
+@@ -504,6 +511,7 @@ TEST_F(RecentTabsSubMenuModelTest, MaxSessionsAndRecency) {
+   // Expected menu items:
+   constexpr ModelData kData[] = {
+       {ui::MenuModel::TYPE_COMMAND, true},    // History
++      CLEAR_BROWSING_DATA_COMMAND_MODEL_DATA
```

Comment:
instead of these changes I *think* you should be able to add `VeryfyModel` method into `RecentTabsSubMenuModelTest`, fix the input data and then call the anonymous-ns version via `::VerifyModel` call.

Reply:
to solve the submenu problem, we should add two methods:
```
  void VerifyModel(const RecentTabsSubMenuModel& model, base::span<const ModelData> data) {
    // do changes only here
    ::VerifyModel(model, data);
  }

  void VerifyModel(const ui::MenuModel* model, base::span<const ModelData> data) {
    // keep as is
    ::VerifyModel(model, data);
  }
```

to only be called for the root menu model.

Reply:
Thanks for the tip! Updated.